### PR TITLE
[ISSUE #2822] Use retainAll method instead of iterating over keys in Map [EventMeshRecommendImpl]

### LINKED ISSUE #2822: [Enhancement] Use retainAll method instead of iterating over keys in Map [EventMeshRecommendImpl]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

located at:
eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java line 176

Iteration over keyset can be avoided.

### Describe the solution you'd like

Using the retainAll method to get the intersection of  sets from the keySet method of the Map interface to get the set of keys for both map1 and map2 can avoid iteration over entire keySet.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mvadharia**
Fixes #2822 .

### Motivation

*Iteration over keyset can be avoided or use of containsAll could be avoided*

### Modifications

*Using the retainAll method to get the intersection of sets from the keySet method of the Map interface to get the set of keys for both map1 and map2 can avoid iteration over entire keySet.*

### Documentation

- Does this pull request introduce a new feature? ( no)

### CODE REVIEW

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java [hunk 1]
```diff
@@ -226,15 +228,16 @@ private String recommendProxy(final Map<String, String> eventMeshMap,
             log.info("eventMeshMap:{},clientDistributionMap:{},group:{}", eventMeshMap, clientDistributionMap, group);
         }
 
-        if (!eventMeshMap.keySet().containsAll(clientDistributionMap.keySet())) {
+        Set<String> clientdDistributionSet=new HashSet<>(clientDistributionMap.keySet());
+        clientdDistributionSet.retainAll(eventMeshMap.keySet());
+        if(!clientdDistributionSet.isEmpty()){
```
Comment (jonyangx) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java:233]: The retainAll() method of java.util.Set interface is used to retain from this set all of its elements that are contained in the specified collection.
I think the code logic is invalid.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java [hunk 2]
```diff
@@ -226,15 +228,16 @@ private String recommendProxy(final Map<String, String> eventMeshMap,
             log.info("eventMeshMap:{},clientDistributionMap:{},group:{}", eventMeshMap, clientDistributionMap, group);
         }
 
-        if (!eventMeshMap.keySet().containsAll(clientDistributionMap.keySet())) {
+        Set<String> clientdDistributionSet=new HashSet<>(clientDistributionMap.keySet());
+        clientdDistributionSet.retainAll(eventMeshMap.keySet());
```
Comment (jonyangx) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java:232]: Retains only the elements in this collection that are contained in the specified collection (optional operation). In other words, removes from this collection all of its elements that are not contained in the specified collection.
This implementation iterates over this collection, checking each element returned by the iterator in turn to see if it's contained in the specified collection. If it's not so contained, it's removed from this collection with the iterator's remove method.
I think Iteration over keyset can not be avoided .

Comment (mvadharia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java:232]: @jonyangx  Idea was to use retainAll() method to get intersection of two sets of keys, if there exists one element in intersection set which indirectly proves there exist proxy(key) not register(i.e. not present in eventMeshMap) but exist in distributionMap"

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java (hunk 1 above)
Comment (mvadharia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java:233]: Need to invert it to get correct logic when using retainAll()

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java (hunk 2 above)
Comment (jonyangx) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java]: public boolean removeAll(Collection<?> c) {
        Objects.requireNonNull(c);
        boolean modified = false;
        Iterator<?> it = iterator();
        while (it.hasNext()) {
            if (c.contains(it.next())) {
                it.remove();
                modified = true;
            }
        }
        return modified;
    }

The removeAll method also uses iterator()  , and use remove operation.  Why not use containAll directly?

Comment (mvadharia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java]: Ok. Got it. Then would close the PR and issue.

**Pull request closed without merging**